File: lib/models/expense.dart
class Expense {
  final String id;
  final String userId;
  final String categoryId;
  final String? groupId;
  final String? paymentMethodId;
  final double amount;
  final String currency;
  final DateTime date;
  final String? note;
  final String? imagePath;
  final DateTime createdAt;
  final DateTime updatedAt;

  Expense({
    required this.id,
    required this.userId,
    required this.categoryId,
    this.groupId,
    this.paymentMethodId,
    required this.amount,
    required this.currency,
    required this.date,
    this.note,
    this.imagePath,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/src/main/java/com/walletapp/demo/entities/Expense.java
package com.walletapp.demo.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "expenses", indexes = {
        @Index(name = "idx_expense_id", columnList = "user_id"),
        @Index(name = "idx_expense_category_id", columnList = "category_id"),
        @Index(name = "idx_expense_group_id", columnList = "group_id"),
        @Index(name = "idx_expense_payment_method_id", columnList = "payment_method_id"),

})
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Expense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "expense_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "group_id")
    private Group group;

    @ManyToOne
    @JoinColumn(name = "payment_method_id")
    private PaymentMethod paymentMethod;

    @Column(nullable = false)
    private double amount;

    @Column(nullable = false)
    private String currency;

    private LocalDateTime date;
    private String note;

    @Column(name = "image_path")
    private String imagePath;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

}
